queue/abstract.lua

```diff
@@ -297,7 +297,11 @@ function tube.drop(self)
 
     local space_name = tube[3]
 
-    box.space[space_name]:drop()
+    if self.raw.drop ~= nil then
+        self.raw:drop()
+    else
+        box.space[space_name]:drop()
```

Comment:
```suggestion
        local space_name = tube[3]
        box.space[space_name]:drop()
```
and remove the one above.